src/dotnet/CarbonAware.DataSources.WattTime.Tests/WattTimeDataSourceTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿using CarbonAware.Tools.WattTimeClient;
+using CarbonAware.Tools.WattTimeClient.Model;
+using Castle.Core.Logging;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CarbonAware.DataSources.WattTime.Tests;
+
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
```

Comment:
Why are we suppressing this warning?

Reply:
It's a nuisance warning.  The objects are initialized in the "Setup" method, rather than the constructor, so the compiler isn't smart enough to realize that the warning doesn't apply.  i.e. Setup IS the constructor method, but not to the compiler.  To make it go away "properly" you end up with a bunch of nullable objects that you have to deal with that add nothing but pain and complexity.